Sources/SwiftParser/Expressions.swift

```diff
@@ -2222,7 +2239,15 @@ extension Parser {
   ) -> RawSwitchExprSyntax {
     let (unexpectedBeforeSwitchKeyword, switchKeyword) = self.eat(switchHandle)
 
-    let subject = self.parseExpression(.basic)
+    // if there is no expression, like `switch { default: return false }` then left brace will be parsed as
+    // an `RawClosureExprSyntax`.
```

Comment:
Just an idea to make this clearer.

```suggestion
    // if there is no expression, like `switch { default: return false }` then left brace would parsed as
    // a `RawClosureExprSyntax` in the condition, which is most likely not what the user meant. 
    // Create a missing condition instead and use the `{` for the start of the body.
```

Same below